include/boost/json/basic_parser.hpp

```diff
@@ -296,103 +354,86 @@ parse_comment(
     ++cs;
 do_com1:
     if(BOOST_JSON_UNLIKELY(! cs))
-    {
-        if(BOOST_JSON_LIKELY(more_))
-            suspend(state::com1);
-        return result::partial;
-    }
+        return partial_if_more(cs.begin(), state::com1);
     switch(*cs)
     {
     case '/':
         ++cs;
 do_com2:
         remain = cs.remain();
-        end_tok = remain ? static_cast<const char*>(
-            std::memchr(cs.data(), '\n', remain)) : nullptr;
-        if(BOOST_JSON_UNLIKELY(! end_tok))
+        cs = remain ? static_cast<const char*>(
+            std::memchr(cs.begin(), '\n', remain)) : nullptr;
```

Comment:
```
if( remain > 0 )
  cs = static_cast<const char*>(
    std::memchr(cs.data(), '\n', remain) );
else
  cs = nullptr;
```

Reply:
Yeah, I'll be moving this out of the function.